kratos/utilities/auxiliar_model_part_utilities.cpp

```diff
@@ -24,6 +23,122 @@
 
 namespace Kratos
 {
+void AuxiliarModelPartUtilities::AddElementWithNodes(
+    Element::Pointer pNewElement,
+    IndexType ThisIndex
+    )
+{
+    const auto& r_geom = pNewElement->GetGeometry();
+    std::vector<IndexType> list_of_nodes(r_geom.size());
+    for (IndexType i = 0; i < r_geom.size(); ++i) {
+        list_of_nodes[i] = r_geom[i].Id();
+    }
+    mrModelPart.AddNodes(list_of_nodes);
+    mrModelPart.AddElement(pNewElement);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void AuxiliarModelPartUtilities::AddElementsWithNodes(
+    std::vector<IndexType> const& rElementIds,
+    IndexType ThisIndex
+    )
+{
+    KRATOS_TRY
+    mrModelPart.AddElements(rElementIds, ThisIndex);
+    if(mrModelPart.IsSubModelPart()) { // Does nothing if we are on the top model part
```

Comment:
This PR is quite old, so I don't remember what I was doing, I will check

Reply:
Okay, I checked, we do nothing in the root modelpart because the root modelpart always have the nodes. Remember that this PR is for adding a method that when you add the elements/conditions will also add the corresponding the nodes contained in those elements. I will add a warning otherwise. And I will reduce the code duplication, as you mentioned.

Reply:
I have not finished, I will finish on Monday, but I push because I am doing the changes from an external machine.

Reply:
I reduced code duplication and took into account your suggestions @pooyan-dadvand

Reply:
So it cannot be used to populate an empty root modelpart? Then I would protect it in some way. @KratosMultiphysics/technical-committee any suggestion?

Reply:
Thinking more about it and we can leave it as it is until we find it error prone.